pkg/injector/webhook.go

```diff
@@ -344,7 +344,12 @@ func createOrUpdateMutatingWebhook(clientSet kubernetes.Interface, cert *certifi
 				return err
 			}
 
-			mwhc.ObjectMeta = existing.ObjectMeta // copy the object meta which includes resource version, required for updates.
+			// copy the object meta which includes resource version, required for updates.
+			// but use the OSM Version, which might be different if this is an upgrade
```

Comment:
It would be nice to have an example in the comments to make the comment clearer.

Reply:
something like ?
```suggestion
			// copy the object meta which includes resource version, required for updates.
			// but use the OSM Version, which might be different if this is an upgrade.  For example, 
			// the webhook might have been initially created by version 1.1.2 and the version osm is 1.2.2
```

Reply:
It was a bit hard for me to understand the following:
`required for updates`: what updates? (This is an existing comment prior to your change, but seems ambiguous)
`but use the OSM Version which might be different if this is an upgrade`: It's not clear what upgrade this is referring to - k8s, control plane, etc.

Just reading the comments, I found it hard to make sense of the relation between object meta and OSM version, in this conext.

Reply:
oh yes that makes sense.  Let me see if I can come up with something clearer

Reply:
Thanks. This comment isn't blocking, so let me know if you want to merge this as is.